drivers/hmis/app/models/hmis/hud/concerns/service_history_queuer.rb

```diff
@@ -0,0 +1,26 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis::Hud::Concerns::ServiceHistoryQueuer
+  extend ActiveSupport::Concern
+
+  included do
+    # Queue up service history enrollment job. This should happen when certain HUD records change.
+    #
+    # If job is already queued, do nothing
+    # If job is currently running, queue it to run 5 minutes from now.
+    # Otherwise, queue the job.
+    def queue_service_history_processing!
+      handlers = ['GrdaWarehouse::Tasks::ServiceHistory::Enrollment', 'batch_process_unprocessed!']
+      return if Delayed::Job.queued?(handlers)
+
+      currently_running = Delayed::Job.running?(handlers)
+      run_at = currently_running ? 5.minutes.from_now : nil
```

Comment:
maybe?
```suggestion
      run_at = currently_running ? 5.minutes.from_now : Time.current
```

Reply:
What happens when that time is slightly in the past, though? I wasn't sure, so I figured it's safer to give it the default behavior.

Reply:
default behavior is good.  If the time is in the past it assumes it should be run.  I just was unclear if passing a nil was the same as not passing the argument in this case.

Reply:
gotcha. I looked at the code and they should both be the same